### PR TITLE
feat: add fly.http.serveStatic

### PR DESCRIPTION
**Pull request by pudility**
Allows for files to be served statically (#66). Example:
```JavaScript
fly.http.serveStatic('public')
```
This will be much better once globs work.

### CODE REVIEW

**Comment by mrkurt** [comment]
This shouldn't be a global, it would be better as a function generator you could use something like this:

```javascript
const fetchFiles = fly.http.serveStatic("public")
fly.http.respondWith(function(req){
  return fetchFiles(req)
})
```

This is a good example of how to do that: https://github.com/superfly/onehostname/blob/master/src/backends.ts

**Comment by pudility** [comment]
@mrkurt I am fine with doing that. Just so you know, normal respondWith still works. You can use them together like this:
```JavaScript
fly.http.serveStatic('public')

fly.http.respondWith(async (req) => new Response('hello'))
```

**Comment by mrkurt** [comment]
Yeah I saw that! I just want to (a) prevent globals and (b) use the conventions we're establishing with other stuff. Lot's of this is just in our heads at the moment, so hard to figure out from scratch.

**Comment by pudility** [comment]
Sounds good. I will change it to what you suggested then.

**Comment by pudility** [comment]
I was just thinking, should there also be some sort of method that returns weather a path should be routed to the static file fetcher or to something else? Because otherwise you still need to manually implement an if statement to check the path.

**Comment by mrkurt** [comment]
I think that's the code we want people to write! People want to do all kinds of routing logic — something like this will try a fetch and then try a different one if it 404s:

```javascript
const blog = backends.ghost("fly-io")
const static = fly.http.serveStatic("public")
fly.http.respondWith(async function(req){
  let resp = await blog(req)

  if(resp.status == 404){
      resp = await static(req)
  }
  return resp
});
```

The onehostname demo "mounts" different backends on different paths: https://github.com/superfly/onehostname/blob/master/index.js#L15-L47

Also one other thing: this needs tests. :D

**Comment by pudility** [comment]
I understand much better now. I will make those changes after lunch.

**Comment by pudility** [comment]
I was working on a test case but, for some reason I am getting a 500 status back when I test it. @kurt Would you mind looking over this? I can't seem to figure it out and I am sure its something really simple. I have the `static-files` directory set up as a fly server - it works when I run it with node but, errors when I run the test.

### static-file.spec.ts
```JavaScript
import { expect } from 'chai'
import { startServer, stopServer } from './helper'
import axios from 'axios'

describe('Static Files', function () {
  before(startServer('static-files'))
  after(stopServer)

  it('fetches dummy text', async () => {
    let res = await axios.get("http://127.0.0.1:3333/public/dummy.txt", { headers: { host: "test" } })

    expect(res.status).to.equal(200)
    expect(res.data).to.equal("foo bar")
  })
})
```
### Error
```
1) Static Files
       fetches dummy text:

      AssertionError: expected 500 to equal 200
```

File: apps/serve-static/index.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+fly.http.serveStatic('public')
```
Comment (jeromegn) [inline apps/serve-static/index.js]: I don't think that still works, does it?

File: v8env/fly/http.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 let router = null
 let fetchEventBound = false
 let flyFetchHandler = null
+let staticDirectory = null
```
Comment (jeromegn) [inline v8env/fly/http.js]: That's not used anymore, I think.

**Comment by jeromegn** [comment]
Closed in favour of #107

**Pull request closed without merging**